cmd/web3/accounts.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"math/big"
+	"strings"
+
+	"github.com/gochain-io/web3"
+	"github.com/urfave/cli"
+)
+
+func getAddressHash(addrHash, privateKey string) string {
+	if addrHash == "" {
+		if privateKey == "" {
+			fatalExit(errors.New("Missing address. Command requires address or private key argument."))
+		}
+		acct, err := web3.ParsePrivateKey(privateKey)
+		if err != nil {
+			fatalExit(err)
+		}
+		addrHash = acct.PublicKey()
+	}
+	return addrHash
+}
+func GetAddressDetails(ctx context.Context, network web3.Network, addrHash, privateKey string) {
+	addrHash = getAddressHash(addrHash, privateKey)
+	client, err := web3.NewClient(network.URL)
+	if err != nil {
+		fatalExit(fmt.Errorf("Failed to connect to %q: %v", network.URL, err))
+	}
+	defer client.Close()
+	bal, err := client.GetBalance(ctx, addrHash, nil)
+	if err != nil {
+		fatalExit(fmt.Errorf("Cannot get address balance from the network: %v", err))
+	}
+	code, err := client.GetCode(ctx, addrHash, nil)
+	if err != nil {
+		fatalExit(fmt.Errorf("Cannot get address code from the network: %v", err))
+	}
+	if verbose {
+		log.Println("Address details:")
+	}
+
+	switch format {
+	case "json":
+		data := struct {
+			Balance *big.Int `json:"balance"`
+			Code    *string  `json:"code"`
+		}{Balance: bal}
+		if len(code) > 0 {
+			sc := string(code)
+			data.Code = &sc
+		}
+		fmt.Println(marshalJSON(&data))
+		return
+	}
+
+	fmt.Println("Balance:", web3.WeiAsBase(bal), network.Unit)
+	if len(code) > 0 {
+		fmt.Println("Code:", string(code))
+	}
+}
+
+func GetBalance(ctx context.Context, c *cli.Context, network web3.Network) {
+	addrHash := getAddressHash(c.String("address"), c.String("private-key"))
+	token := strings.ToUpper(c.String("token")) // preparing for more
+	if token != "GO" {
+		fatalExit(fmt.Errorf("Token %v not supported", token))
+	}
+	client, err := web3.NewClient(network.URL)
```

Comment:
Are we forcing them to pass `-token GO`, even when they want to look up an ETH balance?

Reply:
`network.Unit` will contain the token name for known networks

Reply:
No, the idea is to you'd pick the token and we'd have a list of them available. eg:

```sh
web3 balance --addr ABC --token GO
web3 balance --addr ABC --token ETH
web3 balance --addr ABC --token BNB
```

I get the network thing throws that off right now, but once we have the list, it would also have the associated network and contract address to use.

Reply:
Ok, well this code as written does force them to pass `-token GO` in order to check an eth network balance. We already have the list for supported networks. Why not compare the token to `network.Unit`? Better yet, why not allow for empty token, and let that imply the base unit of the network they are connecting to?

Reply:
If you are planning to look up the network from the token symbol, then that makes the unique token problem even worse, because they have to be unique across all networks (including testnets).

Reply:
Regarding passing the flag, they don't have to currently, it defaults to GO silently so it will work with ETH or GO already. 

The uniqueness isn't a problem because testnets won't be included in this list, and I'm only talking about official coins, ie: the ones on coinmarketcap and/or exchanges, which **are** unique across all chains.

Reply:
Can we establish a `type Token struct{ Network, Address string }` which is either implied by the token flag, or built from the other params, but never a combination of both?

Reply:
The trouble with rejecting that ^ command is that you don't know if the network was set from the flag or the environment variable. Silently overriding an explicit flag seems more severe than silently overriding an env var (definitely wrong intent, v.s. leftover state).

Reply:
Agree, it's either token alone or the other params, thought I had a comment somewhere in there about that.